### PR TITLE
dev, v3.0, v2.1: fix broken links

### PR DESCRIPTION
**Pull request by CaitinChen**
### What is changed, added or deleted? 

Fix broken links.

### Which version does your change affect? 

dev, v3.0, v2.1

### Checklist 

- [x] Add a new file to `TOC.md`
- [x] No Tab spaces anywhere 
- [x] Leave a blank line both before and after a code block
- [x] Keep the first level heading consistent with `title` in metadata
- [x] Use *four* spaces for each level of indentation except that in `TOC.md`

@lilin90 PTAL

### CODE REVIEW

File: dev/how-to/deploy/orchestrated/ansible.md [hunk 1]
```diff
@@ -20,8 +20,8 @@ You can use the TiDB-Ansible configuration file to set up the cluster topology a
 - Deploy the whole TiDB cluster
 - [Start the TiDB cluster](/how-to/deploy/orchestrated/ansible-operations.md#start-a-cluster)
 - [Stop the TiDB cluster](/how-to/deploy/orchestrated/ansible-operations.md#stop-a-cluster)
-- [Modify component configuration](/how-to/maintain/upgrade/rolling-updates-with-ansible.md#modify-component-configuration)
-- [Scale the TiDB cluster](/how-to/maintain/scale/with-ansible.md)
+- [Modify component configuration](/how-to/upgrade/rolling-updates-with-ansible.md#modify-component-configuration)
+- [Scale the TiDB cluster](/how-to/scale/with-ansible/#scale-the-tidb-cluster-using-tidb-ansible.md)
```
Comment (lilin90) [inline dev/how-to/deploy/orchestrated/ansible.md]:
```suggestion
- [Scale the TiDB cluster](/how-to/scale/with-ansible.md)
```

File: v2.1/how-to/deploy/orchestrated/ansible.md [hunk 2]
```diff
@@ -20,10 +20,10 @@ You can use the TiDB-Ansible configuration file to set up the cluster topology a
 - Deploy the whole TiDB cluster
 - [Start the TiDB cluster](/how-to/deploy/orchestrated/ansible-operations.md#start-a-cluster)
 - [Stop the TiDB cluster](/how-to/deploy/orchestrated/ansible-operations.md#stop-a-cluster)
-- [Modify component configuration](/how-to/maintain/upgrade/rolling-updates-with-ansible.md#modify-component-configuration)
-- [Scale the TiDB cluster](/how-to/maintain/scale/with-ansible.md)
+- [Modify component configuration](/how-to/upgrade/rolling-updates-with-ansible.md#modify-component-configuration)
+- [Scale the TiDB cluster](/how-to/scale/with-ansible/#scale-the-tidb-cluster-using-tidb-ansible.md)
```
Comment (lilin90) [inline v2.1/how-to/deploy/orchestrated/ansible.md]:
```suggestion
- [Scale the TiDB cluster](/how-to/scale/with-ansible.md)
```

File: v3.0/how-to/deploy/orchestrated/ansible.md [hunk 3]
```diff
@@ -21,10 +21,10 @@ You can use the TiDB-Ansible configuration file to set up the cluster topology a
 - Deploy the whole TiDB cluster
 - [Start the TiDB cluster](/how-to/deploy/orchestrated/ansible-operations.md#start-a-cluster)
 - [Stop the TiDB cluster](/how-to/deploy/orchestrated/ansible-operations.md#stop-a-cluster)
-- [Modify component configuration](/how-to/maintain/upgrade/rolling-updates-with-ansible.md#modify-component-configuration)
-- [Scale the TiDB cluster](/how-to/maintain/scale/with-ansible.md)
+- [Modify component configuration](/how-to/upgrade/rolling-updates-with-ansible.md#modify-component-configuration)
+- [Scale the TiDB cluster](/how-to/scale/with-ansible/#scale-the-tidb-cluster-using-tidb-ansible.md)
```
Comment (lilin90) [inline v3.0/how-to/deploy/orchestrated/ansible.md]:
```suggestion
- [Scale the TiDB cluster](/how-to/scale/with-ansible.md)
```

File: dev/how-to/upgrade/rolling-updates-with-ansible.md [hunk 4]
```diff
@@ -16,8 +16,8 @@ When you perform a rolling update for a TiDB cluster, the service is shut down s
 
 - To upgrade between large versions, you must upgrade [`tidb-ansible`](https://github.com/pingcap/tidb-ansible).
 
-    - To upgrade the TiDB version from 1.0 to 2.1, see [TiDB 2.1 Upgrade Guide](/how-to/upgrade/to-tidb-2.1.md).
-    - To upgrade the TiDB version from 2.1 to 3.0, see [TiDB 3.0 Upgrade Guide](/how-to/upgrade/from-previous-version.md).
+    - To upgrade TiDB 2.0 (TiDB 2.0.1 or later) or TiDB 2.1 RC version to TiDB 2.1 GA version, see [TiDB 2.1 Upgrade Guide](/v2.1/how-to/upgrade/from-previous-version.md).
+    - To upgrade TiDB 2.0 (above V2.0.1) or TiDB 2.1 RC to TiDB 3.0, see [TiDB 3.0 Upgrade Guide](/how-to/upgrade/from-previous-version.md).
```
Comment (lilin90) [inline dev/how-to/upgrade/rolling-updates-with-ansible.md]:
```suggestion
    - To upgrade TiDB 2.0 (TiDB 2.0.1 or later) or TiDB 2.1 RC to TiDB 3.0, see [TiDB 3.0 Upgrade Guide](/how-to/upgrade/from-previous-version.md).
```

File: v3.0/how-to/upgrade/rolling-updates-with-ansible.md [hunk 5]
```diff
@@ -17,8 +17,8 @@ When you perform a rolling update for a TiDB cluster, the service is shut down s
 
 - To upgrade between large versions, you must upgrade [`tidb-ansible`](https://github.com/pingcap/tidb-ansible).
 
-    - To upgrade the TiDB version from 1.0 to 2.1, see [TiDB 2.1 Upgrade Guide](/how-to/upgrade/to-tidb-2.1.md).
-    - To upgrade the TiDB version from 2.1 to 3.0, see [TiDB 3.0 Upgrade Guide](/how-to/upgrade/from-previous-version.md).
+    - To upgrade TiDB 2.0 (TiDB 2.0.1 or later) or TiDB 2.1 RC version to TiDB 2.1 GA version, see [TiDB 2.1 Upgrade Guide](/v2.1/how-to/upgrade/from-previous-version.md).
+    - To upgrade TiDB 2.0 (above V2.0.1) or TiDB 2.1 RC to TiDB 3.0, see [TiDB 3.0 Upgrade Guide](/how-to/upgrade/from-previous-version.md).
```
Comment (lilin90) [inline v3.0/how-to/upgrade/rolling-updates-with-ansible.md]:
```suggestion
    - To upgrade TiDB 2.0 (TiDB 2.0.1 or later) or TiDB 2.1 RC to TiDB 3.0, see [TiDB 3.0 Upgrade Guide](/how-to/upgrade/from-previous-version.md).
```

**Comment by CaitinChen** [comment]
@lilin90 Thanks. PTAL again.

**Pull request merged**